inbm/dispatcher-agent/dispatcher/sota/downloader.py

```diff
@@ -165,3 +168,66 @@ def download(self,
 
     def check_release_date(self, release_date: Optional[str]) -> bool:
         return self.is_valid_release_date(release_date)
+
+
+class TiberOSDownloader(Downloader):
+    """TiberOSDownloader class, child of Downloader
+
+       @param signature: signature used to preform signature check on downloaded image.
+    """
+
+    def __init__(self, signature: Optional[str] = None) -> None:
+        super().__init__()
+        self._signature = signature
+
+    def download(self,
+                 dispatcher_broker: DispatcherBroker,
+                 uri: Optional[CanonicalUri],
+                 repo: IRepo,
+                 username: Optional[str],
+                 password: Optional[str],
+                 release_date: Optional[str]) -> None:
+        """Downloads files and places image in local cache
+
+        @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+        @param uri: URI of the source location
+        @param repo: repository for holding the download
+        @param username: username to use for download
+        @param password: password to use for download
+        @param release_date: manifest release date
+        @raises SotaError: release date is not valid
+        """
+
+        if uri is None:
+            raise SotaError("URI is None while performing TiberOS download")
+
+        if password is None:
+            raise SotaError("JWT Token is None while performing TiberOS download")
+
+        oras_download(dispatcher_broker=dispatcher_broker,
+                      uri=uri,
+                      repo=repo,
+                      umask=UMASK_OTA,
+                      username=username,
+                      password=password)
+
+        # Perform signature check.
+        # Multiple files may have been downloaded from OCI.
+        # The method below will iterate over all files in the repo, calculate the SHA256sum for each file,
+        # and compare it with the provided signature.
```

Comment:
this returns if any one of the files match the signature--this could be a problem if the desired file is not the one we're checking. is there any way to know which file we're supposed to check here?

Reply:
Unfortunately no, it will pull all the artifacts store in that registry. For example, the oras pull below pull the checksum file and raw file.

Downloading fe1bb328ca88 tiber-readonly-2.0.20240919.0103.raw.xz.sha256sum
Downloading 64a37255b4eb tiber-readonly-2.0.20240919.0103.raw.xz
Downloaded  fe1bb328ca88 tiber-readonly-2.0.20240919.0103.raw.xz.sha256sum
Downloaded  64a37255b4eb tiber-readonly-2.0.20240919.0103.raw.xz
Pulled [registry] registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest
Digest: sha256:d3fbad0d4d25cb96f6f3ee3e47545334d3612cb8ad4af63c25137f18c47047c9

Reply:
This should be okay then; the signature is designed to check the file contents. If the filename doesn't matter, then we still have a verified blob to work with.